File: Python/Arrays&Hashing/TwoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        # Solution 1 ~ Brute Force ~ WC TC: O(n)
        '''
        for i, numi in enumerate(nums):
            for j, numj in enumerate(nums):
                if i != j and numi + numj == target:
                    return [min(i,j),max(i,j)]
        '''
        
        # Solution 2 - Hash Map
        prevMap = {} # val : index
        for i, n in enumerate(nums):
            diff = target - n
            if diff in prevMap:
                return [prevMap[diff], i]
            prevMap[n] = i
        return

File: Go/Arrays&Hashing/TwoSum.go
func twoSum(nums []int, target int) []int {
    // Solution 1 - Brute Force
    /*
    res := make([]int, 0)
    for i, numi := range(nums) {
        for j, numj := range(nums) {
            if i != j && numi + numj == target {
                res = append(res, i)
                res = append(res, j)
                return res
            }
        }
    }
    return nil
    */
    // Solution 2 - HashMap
    prevMap := make(map[int]int)
    for i, n := range(nums) {
        if val, found := prevMap[target-n]; found {
            return []int{val, i}
        }
        prevMap[n] = i
    }
    return nil
}
